dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -18,59 +19,67 @@ class DummyOutcomeRefuter(CausalRefuter):
+            2. Noise
+            This adds white noise to the outcome with white noise, reducing any causal relationship with the treatment.
+            3. Zero
+            It replaces all the values in the outcome by zero
+
+    The pipeline is of the following form:
+    * If the function pd.Dataframe -> np.ndarray is already defined.
+    [(func,func_params),('permute', permute_fraction), ('noise', std_dev)]
+    * If a function from the above list is used
+    [('knn',{'n_neighbors':5}), ('permute', permute_fraction), ('noise', std_dev)]
+
     - 'required_variables': int, list, bool, True by default
-    A user can input either an integer value, list or bool.
+    The inputs are either an integer value, list or bool.
         1. An integer argument refers to how many variables will be used for estimating the value of the outcome
-        2. A list allows the user to explicitly refer to which variables will be used to estimate the outcome
-            Furthermore, a user can either choose to select the variables desired. Or they can deselect the variables,
-            that they do not want in their analysis. 
+        2. A list explicitly refers to which variables will be used to estimate the outcome
+            Furthermore, it gives the ability to explictly select or deselect the covariates present in the estimation of the 
+            outcome. This is done by either adding or explicitly removing variables from the list as shown below: 
             For example:
             We need to pass required_variables = [W0,W1] is we want W0 and W1.
             We need to pass required_variables = [-W0,-W1] if we want all variables excluding W0 and W1.
         3. If the value is True, we wish to include all variables to estimate the value of the outcome. A False value is INVALID
            and will result in an error. 
-    - scale: float, 1.0 by default
-    The value by which the std_deviation of the dummy outcome has to be scaled 
+    Note:
+    These inputs are fed to the function for estimating the outcome variable. The same set of required_variables is used for each
+    instance of an internal function.
     """
+    # The currently supported estimators
+    SUPPORTED_ESTIMATORS = ["linear_regression", "knn", "svm", "random_forest", "neural_network"]
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._dummy_outcome_type = kwargs.pop("placebo_type", None)
-        if self._dummy_outcome_type is None:
-            self._dummy_outcome_type = "Random Data"
+        
         self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS)
-        self._random_state = kwargs.pop("random_state", None)
-        self._outcome_function = kwargs.pop("outcome_function", None)
-        self._params = kwargs.pop("params", None)
-        self._scale = kwargs.pop("scale", 1)
+        pipeline = kwargs.pop("pipeline",[("zero",""),("noise", 1)])

```

Comment:
for generalizability, instead of zero, you can call it constant, and then its parameter will be 0.